src/components/ContentPreview/ContentPreview.js

```diff
@@ -1078,73 +1083,89 @@ class ContentPreview extends PureComponent<Props, State> {
         /* eslint-disable jsx-a11y/no-noninteractive-tabindex */
         return (
             <Internationalize language={language} messages={messages}>
-                <div
-                    id={this.id}
-                    className={`be bcpr ${className}`}
-                    ref={measureRef}
-                    onKeyDown={this.onKeyDown}
-                    tabIndex={0}
-                >
-                    {hasHeader && (
-                        <Header
-                            file={file}
-                            token={token}
-                            onClose={onClose}
-                            onPrint={this.print}
-                            canDownload={this.canDownload()}
-                            onDownload={this.download}
-                            contentOpenWithProps={contentOpenWithProps}
-                            canAnnotate={this.canAnnotate()}
-                        />
-                    )}
-                    <div className="bcpr-body">
-                        <div className="bcpr-container" onMouseMove={this.onMouseMove} ref={this.containerRef}>
-                            {file ? (
-                                <Measure bounds onResize={this.onResize}>
-                                    {({ measureRef: previewRef }) => <div ref={previewRef} className="bcpr-content" />}
-                                </Measure>
-                            ) : (
-                                <div className="bcpr-loading-wrapper">
-                                    <PreviewLoading
-                                        isLoading={!isFileError}
-                                        loadingIndicatorProps={{
-                                            size: 'large',
-                                        }}
-                                    />
-                                </div>
-                            )}
-                            {hasLeftNavigation && (
-                                <PlainButton type="button" className="bcpr-navigate-left" onClick={this.navigateLeft}>
-                                    <IconNavigateLeft />
-                                </PlainButton>
-                            )}
-                            {hasRightNavigation && (
-                                <PlainButton type="button" className="bcpr-navigate-right" onClick={this.navigateRight}>
-                                    <IconNavigateRight />
-                                </PlainButton>
+                <FeatureProvider features={features}>
```

Comment:
this change just wraps everything with a feature provider

Reply:
Could this use a HOC to avoid the additional wrapper?

Reply:
no, it takes a prop

Reply:
unless you want to implicitly extract the prop from the parent into the provider inside the HOC

Reply:
Yep, that's what I was thinking.